src/index.ts

```diff
@@ -3,7 +3,148 @@
 'use-strict';
 
 import {
+  ICell, ICodeCellViewModel, IMarkdownCellViewModel, IRawCellViewModel
+} from 'jupyter-js-cell';
+
+import {
+  IObservableList
+} from 'phosphor-observablelist';
+
+import {
   Widget
 } from 'phosphor-widget';
 
 import './index.css';
+
+
+/**
+ * The interactivity modes for a notebook.
+ */
+export
+enum NotebookMode {
+  /**
+   * Command mode is used for navigation and manipulation.
+   */
+  Command,
+
+  /**
+   * Edit mode is used for text and code editing.
+   */
+  Edit
+}
+
+
+
+/**
+ * An enum which describes which interface property has changed.
+ */
+export
+enum NotebookProperty {
+  DefaultMimetype,
+  DirtyIndicator,
+  Trusted,
+  Mode,
+  SelectedCell,
+}
+
+
+/**
+ * The arguments object emitted with the `stateChanged` signal.
+ */
+export
+interface INotebookChangedArgs<T> {
```

Comment:
OK I am fine with moving to strings rather than enums.

On Tue, Nov 3, 2015 at 7:48 PM, S. Chris Colbert <notifications@github.com>
wrote:

> In src/index.ts
> <https://github.com/jupyter/jupyter-js-notebook/pull/2#discussion_r43829590>
> :
>
> > + */
> > +export
> > +enum NotebookProperty {
> > +  DefaultMimetype,
> > +  DirtyIndicator,
> > +  Trusted,
> > +  Mode,
> > +  SelectedCell,
> > +}
> > +
> > +
> > +/**
> > + * The arguments object emitted with the `stateChanged` signal.
> > + */
> > +export
> > +interface INotebookChangedArgs<T> {
>
> It can useful if you ever decide to expose a *more specific* signal for a
> particular property.
>
> Also, see here for a comment on why I'm now -1 on using an enum for which
> (should be a string):
> phosphorjs/phosphor-properties#7
> <https://github.com/phosphorjs/phosphor-properties/pull/7>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter-js-notebook/pull/2/files#r43829590>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com